### ISSUE TITLE
Reduce boiler plate code in otel-profiling-agent 

### ISSUE THREAD

**Issue by florianl** [elastic/otel-profiling-agent#89]
The general idea is to reduce code boiler plate, that does not contribute to profiling in general from `otel-profiling-agent`.

Without these named packages it is still possible to provide a minimal agent in `otel-profiling-agent`, reduce the code base and therefore maintenance burden.

| package| Reason|
| -- | -- |
| `config` | The purpose of `otel-profiling-agent` is to provide a minimal proof of concept agent. The capabilities of package `config` originates from a long history of optimyze and complexity to deal with the Elastic environment. |
| `containermetadata` | With [OTel Resources](https://opentelemetry.io/docs/specs/semconv/resource/) the OTel environment does have a different view on things like virtualization. containermetadata does not cover [OTel Resources](https://opentelemetry.io/docs/specs/semconv/resource/) and vice versa. |
| `host` | Package `host` [is about](https://github.com/elastic/otel-profiling-agent/blob/main/host/host.go) `TraceHash` and `FileID` - its content should be in `libpf`. The package name `host` suggests something the package is not about. |
| `hostmetadata` | The annotations by `hostmetadata` are nice but conflict with the view of OTel on [OTel Resources](https://opentelemetry.io/docs/specs/semconv/resource/). Making them OTel compliant adds burden and cause compatibility issues. |
| ~~`metrics`~~ | ~~Similar to `containermetadata` and `hostmetadata`, metrics originates from a different reasoning and fulfills a different purpose.~~ |
| `platform` | Similar to `containermetadata` and `hostmetadata` and `metrics`. |

The overall idea originates from the understanding, that people will use `otel-profiling-agent` more as a library rather as a fully fledged agent. The proposed changes should also help keeping complexity low when working on OTel collector receiver parts.

**Comment by brancz** [comment]
We can do without any of these packages (except `host` being integrated elsewhere of course).

That said, I think it does make sense to aim for the `otel-profiling-agent` to be a useful standalone project as well, and not just as a library used elsewhere. I don't know what the right balance here is, but I'd say it should err on the side of integrating as natively into being an otel component as possible.

**Comment by felixge** [comment]
@brancz +1. But in order to be a useful standalone component some degree of host/container meta data support would still be useful, right?

**Comment by florianl** [comment]
@felixge when reducing metadata for the standalone component, I'm thinking about limiting it to information that can be provided by the Go standard library, without external dependencies. So metadata from environments like, GCE, EC2, k8s and others, will be removed. The standalone agent should focus and showcase profiling capabilities and not being a fully fledged agent for every use case.

To use the profiling capabilities in the OTel environment, [there is work](https://github.com/elastic/otel-profiling-agent/pull/87) to integrate profiling via OTel receiver into the OTel collector. People that are interested in using the profiling functionality in their own agents/ways, probably already do have some kind of metadata handling.

And I agree with @brancz that there should be a way to metadata with profiling information. And as pointed out in https://github.com/elastic/otel-profiling-agent/pull/90#discussion_r1681105143 one way to connect these bits can be by using the PID in the `reporter` interface.

**Comment by brancz** [comment]
All I'm saying is we need to make sure there is enough metadata so the otel collector can resolve remaining metadata a user might want to add (I understand this is the division of responsibilities in otel). The collector folks should be able to answer this.

**Comment by florianl** [comment]
After some refactoring was merged, I had to update https://github.com/elastic/otel-profiling-agent/pull/90. Now it is open for review.

It can be summarized that there is no decrease in functionality or change in behavior of the stand alone agent after the removal of the code.

Despite the initial comment of this issue, I did not remove package `metrics`. Package `metrics` reports information from the eBPF programs and internals, that can be essential to debug. Removing it would result in a decrease of functionality.

As pointed out in https://github.com/elastic/otel-profiling-agent/pull/90#discussion_r1681105143 the interface `reporter.ReportCountForTrace()` was updated to let other implementations enrich profiling information based on the PID. This change does not impact the OTel profiling signal, but opens up to other `reporter` interface implementations - like OTel enrichtment processors.
